### ISSUE TITLE
Array comparison breaks if using the environments plugin (until app is restarted!)

### ISSUE THREAD

**Issue by gramatus** [github/safe-settings#681]
## Problem Description
I am trying to figure out safe-settings and think I might have found a bug when updating team permissions. This might be a red herring, but it seems to me that if any teams already  has either direct or indirect access to the repo, then some logic ends up deciding that there are no changes. I've tried a few different combinations of direct and indirect access, and it seems to behave the same way always.

### What is actually happening
My repo has the following access (the two teams was actually added by safe-settings when only my user had access):
![{19A2E25D-B935-4EF5-9FCB-9E612879DD4D}](https://github.com/user-attachments/assets/7153fd93-f479-4e89-9abe-4ddf4adce970)

I have updated the config to add another team and change one access level:
```yaml
repository:
  name: experiments
teams:
  - name: admins
    permission: push
  - name: "admin-codeowner"
    permission: maintain
  - name: "admin-codeowner-github"
    permission: pull
```

To begin with, the PR seems to report the expected results a bit strange:
![{687A990F-3A9C-46EC-B146-ACBF342E556A}](https://github.com/user-attachments/assets/41dd3d35-aa43-4621-a801-c66bd87691aa)

The check details also seems a bit suspect (since it does not say "no changes detected", but neither reports any changes):
![{750DB47E-3121-4BC3-8163-CE84129E4130}](https://github.com/user-attachments/assets/5407d209-67d4-4509-8db6-d51a703f4036)

The logs  also are a bit strange. I've formatted (and simplified) the most interesting bit. After this comes `There are no changes for Teams for repo [object Object]. Skipping changes`.
```json
Results of comparing Teams diffable target
[
    {
        "name": "admin-codeowner",
        "id": 10981460,
        "slug": "admin-codeowner",
        "permission": "pull",
    },
    {
        "name": "admins",
        "id": 10280360,
        "slug": "admins",
        "permission": "push",
    }
]
with source
[
    {
        "name": "admins",
        "permission": "push"
    },
    {
        "name": "admin-codeowner",
        "permission": "maintain"
    },
    {
        "name": "admin-codeowner-github",
        "permission": "pull"
    }
]
is
[object Object]
```

Some more context:

```
2024-09-13T17:21:01.0133575Z {"level":20,"time":1726248061012,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"8a081880-71f4-11ef-8fc2-2bf2ffe375cb","msg":"Finding teams for helse-sorost/experiments"}
2024-09-13T17:21:01.2379181Z {"level":20,"time":1726248061233,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"octokit","msg":"GitHub request: GET https://api.github.com/repos/helse-sorost/experiments/teams - 200"}
2024-09-13T17:21:01.2380197Z {"level":20,"time":1726248061233,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"8a081880-71f4-11ef-8fc2-2bf2ffe375cb","msg":" [\n  {\n    \"name\": \"admin-codeowner\",\n    \"id\": 10981460,\n    \"node_id\": \"T_kwDOCjTDfM4Ap5BU\",\n    \"slug\": \"admin-codeowner\",\n    \"description\": \"\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10981460\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10981460/repos\",\n    \"permission\": \"pull\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": false,\n      \"triage\": false,\n      \"pull\": true\n    },\n    \"parent\": null\n  },\n  {\n    \"name\": \"admins\",\n    \"id\": 10280360,\n    \"node_id\": \"T_kwDOCjTDfM4AnN2o\",\n    \"slug\": \"admins\",\n    \"description\": \"Organization admins for helse-sorost\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10280360\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admins\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10280360/repos\",\n    \"permission\": \"push\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": true,\n      \"triage\": true,\n      \"pull\": true\n    },\n    \"parent\": null\n  }\n] \n\n [\n  {\n    \"name\": \"admins\",\n    \"permission\": \"push\"\n  },\n  {\n    \"name\": \"admin-codeowner\",\n    \"permission\": \"maintain\"\n  },\n  {\n    \"name\": \"admin-codeowner-github\",\n    \"permission\": \"pull\"\n  }\n] "}
2024-09-13T17:21:01.2380793Z {"level":20,"time":1726248061233,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"8a081880-71f4-11ef-8fc2-2bf2ffe375cb","msg":"Results of comparing Teams diffable target [{\"name\":\"admin-codeowner\",\"id\":10981460,\"node_id\":\"T_kwDOCjTDfM4Ap5BU\",\"slug\":\"admin-codeowner\",\"description\":\"\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10981460\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10981460/repos\",\"permission\":\"pull\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":false,\"triage\":false,\"pull\":true},\"parent\":null},{\"name\":\"admins\",\"id\":10280360,\"node_id\":\"T_kwDOCjTDfM4AnN2o\",\"slug\":\"admins\",\"description\":\"Organization admins for helse-sorost\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10280360\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admins\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10280360/repos\",\"permission\":\"push\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":true,\"triage\":true,\"pull\":true},\"parent\":null}] with source [{\"name\":\"admins\",\"permission\":\"push\"},{\"name\":\"admin-codeowner\",\"permission\":\"maintain\"},{\"name\":\"admin-codeowner-github\",\"permission\":\"pull\"}] is [object Object]"}
2024-09-13T17:21:01.2380843Z {"level":20,"time":1726248061233,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"8a081880-71f4-11ef-8fc2-2bf2ffe375cb","msg":"There are no changes for Teams for repo [object Object]. Skipping changes"}
```

### What is the expected behavior
The `admin-codeowner-github` team is granted access to the repo, and the `admin-codeowner` team access level is changed to `maintain`.

### Error output, if available

These are similar logs to above, but from the non-dry-run (log-level `debug`):

```
2024-09-13T17:24:15.5007364Z {"level":20,"time":1726248255500,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"Finding teams for helse-sorost/experiments"}
2024-09-13T17:24:15.6703147Z {"level":20,"time":1726248255669,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"octokit","msg":"GitHub request: GET https://api.github.com/repos/helse-sorost/experiments/teams - 200"}
2024-09-13T17:24:15.6703928Z {"level":20,"time":1726248255670,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":" [\n  {\n    \"name\": \"admin-codeowner\",\n    \"id\": 10981460,\n    \"node_id\": \"T_kwDOCjTDfM4Ap5BU\",\n    \"slug\": \"admin-codeowner\",\n    \"description\": \"\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10981460\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10981460/repos\",\n    \"permission\": \"pull\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": false,\n      \"triage\": false,\n      \"pull\": true\n    },\n    \"parent\": null\n  },\n  {\n    \"name\": \"admins\",\n    \"id\": 10280360,\n    \"node_id\": \"T_kwDOCjTDfM4AnN2o\",\n    \"slug\": \"admins\",\n    \"description\": \"Organization admins for helse-sorost\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10280360\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admins\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10280360/repos\",\n    \"permission\": \"push\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": true,\n      \"triage\": true,\n      \"pull\": true\n    },\n    \"parent\": null\n  }\n] \n\n [\n  {\n    \"name\": \"admins\",\n    \"permission\": \"push\"\n  },\n  {\n    \"name\": \"admin-codeowner\",\n    \"permission\": \"maintain\"\n  },\n  {\n    \"name\": \"admin-codeowner-github\",\n    \"permission\": \"pull\"\n  }\n] "}
2024-09-13T17:24:15.6704822Z {"level":20,"time":1726248255670,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"Results of comparing Teams diffable target [{\"name\":\"admin-codeowner\",\"id\":10981460,\"node_id\":\"T_kwDOCjTDfM4Ap5BU\",\"slug\":\"admin-codeowner\",\"description\":\"\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10981460\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10981460/repos\",\"permission\":\"pull\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":false,\"triage\":false,\"pull\":true},\"parent\":null},{\"name\":\"admins\",\"id\":10280360,\"node_id\":\"T_kwDOCjTDfM4AnN2o\",\"slug\":\"admins\",\"description\":\"Organization admins for helse-sorost\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10280360\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admins\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10280360/repos\",\"permission\":\"push\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":true,\"triage\":true,\"pull\":true},\"parent\":null}] with source [{\"name\":\"admins\",\"permission\":\"push\"},{\"name\":\"admin-codeowner\",\"permission\":\"maintain\"},{\"name\":\"admin-codeowner-github\",\"permission\":\"pull\"}] is [object Object]"}
2024-09-13T17:24:15.6704899Z {"level":20,"time":1726248255670,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"There are no changes for Teams for repo [object Object]. Skipping changes"}
2024-09-13T17:24:15.6704942Z {"level":20,"time":1726248255670,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"Not run in nop"}
2024-09-13T17:24:15.7528287Z {"level":20,"time":1726248255752,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","msg":"Not working on the default branch, returning..."}
2024-09-13T17:24:15.7537728Z {"level":30,"time":1726248255753,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"server","name":"http","req":{"id":"ffd38d1a-71f4-11ef-9688-b45cd5844cba","method":"POST","url":"/","query":{},"params":{},"headers":{"accept":"*/*","host":"hso-gh-config-app-testhsorhf.azurewebsites.net","user-agent":"GitHub-Hookshot/6415e4f","max-forwards":"10","x-github-delivery":"ffd38d1a-71f4-11ef-9688-b45cd5844cba","x-github-event":"push","x-github-hook-id":"483383178","x-github-hook-installation-target-id":"917219","x-github-hook-installation-target-type":"integration","x-hub-signature":"sha1=b8729e5072cf5428533fc333fdbb7971b7c6161e","x-hub-signature-256":"sha256=3a418c9f72ec8a15a3e84a0919c8ae4fd052e158ce6688a197c38a6fe7d97024","x-arr-log-id":"0bc319ea-6a85-402d-a14c-03e80268495b","client-ip":"140.82.115.112:35726","x-client-ip":"140.82.115.112","disguised-host":"hso-gh-config-app-testhsorhf.azurewebsites.net","x-site-deployment-id":"hso-gh-config-app-testhsorhf","was-default-hostname":"hso-gh-config-app-testhsorhf.azurewebsites.net","x-forwarded-proto":"https","x-appservice-proto":"https","x-arr-ssl":"2048|256|CN=Microsoft Azure RSA TLS Issuing CA 07, O=Microsoft Corporation, C=US|CN=*.azurewebsites.net, O=Microsoft Corporation, L=Redmond, S=WA, C=US","x-forwarded-tlsversion":"1.3","x-forwarded-for":"140.82.115.112:35726","x-original-url":"/","x-waws-unencoded-url":"/","x-client-port":"35726","content-type":"application/json","content-length":"7502"},"remoteAddress":"::ffff:169.254.129.1","remotePort":29019},"res":{"statusCode":200,"headers":{"x-powered-by":"Express"}},"responseTime":13,"msg":"POST / 200 - 13ms"}
2024-09-13T17:24:15.7633477Z {"level":20,"time":1726248255763,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"octokit","msg":"GitHub request: GET https://api.github.com/repos/helse-sorost/experiments/teams - 200"}
2024-09-13T17:24:15.7635083Z {"level":20,"time":1726248255763,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":" [\n  {\n    \"name\": \"admin-codeowner\",\n    \"id\": 10981460,\n    \"node_id\": \"T_kwDOCjTDfM4Ap5BU\",\n    \"slug\": \"admin-codeowner\",\n    \"description\": \"\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10981460\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10981460/repos\",\n    \"permission\": \"pull\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": false,\n      \"triage\": false,\n      \"pull\": true\n    },\n    \"parent\": null\n  },\n  {\n    \"name\": \"admins\",\n    \"id\": 10280360,\n    \"node_id\": \"T_kwDOCjTDfM4AnN2o\",\n    \"slug\": \"admins\",\n    \"description\": \"Organization admins for helse-sorost\",\n    \"privacy\": \"closed\",\n    \"notification_setting\": \"notifications_enabled\",\n    \"url\": \"https://api.github.com/organizations/171230076/team/10280360\",\n    \"html_url\": \"https://github.com/orgs/helse-sorost/teams/admins\",\n    \"members_url\": \"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\n    \"repositories_url\": \"https://api.github.com/organizations/171230076/team/10280360/repos\",\n    \"permission\": \"push\",\n    \"permissions\": {\n      \"admin\": false,\n      \"maintain\": false,\n      \"push\": true,\n      \"triage\": true,\n      \"pull\": true\n    },\n    \"parent\": null\n  }\n] \n\n [\n  {\n    \"name\": \"admins\",\n    \"permission\": \"push\"\n  },\n  {\n    \"name\": \"admin-codeowner\",\n    \"permission\": \"maintain\"\n  },\n  {\n    \"name\": \"admin-codeowner-github\",\n    \"permission\": \"pull\"\n  }\n] "}
2024-09-13T17:24:15.7636270Z {"level":20,"time":1726248255763,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"Results of comparing Teams diffable target [{\"name\":\"admin-codeowner\",\"id\":10981460,\"node_id\":\"T_kwDOCjTDfM4Ap5BU\",\"slug\":\"admin-codeowner\",\"description\":\"\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10981460\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admin-codeowner\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10981460/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10981460/repos\",\"permission\":\"pull\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":false,\"triage\":false,\"pull\":true},\"parent\":null},{\"name\":\"admins\",\"id\":10280360,\"node_id\":\"T_kwDOCjTDfM4AnN2o\",\"slug\":\"admins\",\"description\":\"Organization admins for helse-sorost\",\"privacy\":\"closed\",\"notification_setting\":\"notifications_enabled\",\"url\":\"https://api.github.com/organizations/171230076/team/10280360\",\"html_url\":\"https://github.com/orgs/helse-sorost/teams/admins\",\"members_url\":\"https://api.github.com/organizations/171230076/team/10280360/members{/member}\",\"repositories_url\":\"https://api.github.com/organizations/171230076/team/10280360/repos\",\"permission\":\"push\",\"permissions\":{\"admin\":false,\"maintain\":false,\"push\":true,\"triage\":true,\"pull\":true},\"parent\":null}] with source [{\"name\":\"admins\",\"permission\":\"push\"},{\"name\":\"admin-codeowner\",\"permission\":\"maintain\"},{\"name\":\"admin-codeowner-github\",\"permission\":\"pull\"}] is [object Object]"}
2024-09-13T17:24:15.7636358Z {"level":20,"time":1726248255763,"pid":24,"hostname":"hso-gh-con_922470aeb4","name":"probot","name":"probot","name":"event","id":"fe67ba96-71f4-11ef-88b6-6358a3cea164","msg":"There are no changes for Teams for repo [object Object]. Skipping changes"}
```

### Are you using the hosted instance of probot/settings or running your own?
Running my own.

### If running your own instance, are you using it with github.com or GitHub Enterprise?
github.com

#### Version of probot/settings
Not sure. I'm running the published image of [safe-settings 2.1.11](https://github.com/github/safe-settings/pkgs/container/safe-settings/265710551?tag=2.1.11), but I don't know what probot-version that is.

#### Version of GitHub Enterprise
n/a

**Comment by gramatus** [comment]
I've been trying to debug this, and for some reason I cannot reproduce it when running a dev server. Thus it seems it has to be either the settings for the GitHub App or my hosting that breaks this, not the code itself. I will post back when I find a solution.

**Comment by gramatus** [comment]
I believe I have found an issue in the code at this line:
[environments.js#L22](https://github.com/RBC/github-safe-settings/blob/c9247f5f864222263ee8864c5f18062e156fe067/lib/plugins/environments.js#L22)
```javascript
        // Remove 'name' from filtering list so Environments with only a name defined are processed.
        MergeDeep.NAME_FIELDS.splice(MergeDeep.NAME_FIELDS.indexOf('name'), 1)
```

The way `NAME_FIELDS` is stored, this ends up modifying a global object, resulting in a lot of broken stuff after the "environments" plugin is loaded. I will try to come up with a solution and create a PR.

**Comment by gregnrobinson** [comment]
Hey @torgst I was debugging this too and came to the same conclusion. Below is how I debugged it.

We noticed that if we added teams and permissions to a suborg yaml file and pushed the changes, safe-settings was not applying new teams to the repositories after upgrading the safe settings image.

I built version `2.1.10` and deployed it to my admin repo. `2.1.10` is the latest version before `2.1.11-rc.1` where adding teams and permissions works without issues. I [compared the two releases](https://github.com/github/safe-settings/compare/2.1.10...2.1.11-rc.1) and went through the commits.

I built and deployed `2.1.11-rc.1` alongside `2.1.10` and also created a separate GitHub app for each version (not required but made debugging easier)

I worked backwards reverting any modified application files from the following commits...

- [Bump braces from 3.0.2 to 3.0.3](https://github.com/github/safe-settings/commit/f6c8d436457f91308a84e79754b32ec79aa68314)
- [import eta templates as javascript strings (#648)](https://github.com/github/safe-settings/commit/35b2ac49860bcea8455509638809f31c5befe70c)
- [Environments tolerate concise configuration (#649)](https://github.com/github/safe-settings/commit/19e2691e2c1af133aed4939256e4c82ead1ebde9)

I started reverting the individual commits in my `2.1.11-rc.1`  build based on the order above and as it turns out only when I reverted [Environments tolerate concise configuration (#649)](https://github.com/github/safe-settings/commit/19e2691e2c1af133aed4939256e4c82ead1ebde9) was I able to add and remove teams to suborgs. The other two commits returned the same behaviour. I was able to consistently have teams get added and removed by adding the environment configs from `2.1.10` to my `2.1.11-rc.1 build`.

Looking through [Environments tolerate concise configuration (#649)](https://github.com/github/safe-settings/commit/19e2691e2c1af133aed4939256e4c82ead1ebde9), I couldn't identify anything problematic partly due to the lack of my overall understanding of the embedded logic. Still, the first thing I tried was commenting out [this line](https://github.com/github/safe-settings/blob/19e2691e2c1af133aed4939256e4c82ead1ebde9/lib/plugins/environments.js#L22) from the commit in [lib/plugins/environments.js](https://github.com/github/safe-settings/blob/2.1.11-rc.1/lib/plugins/environments.js) and pushing a new build. To my surprise, this solved the problem. I couldn't understand how changes made only to [lib/plugins/environments.js](https://github.com/github/safe-settings/blob/2.1.11-rc.1/lib/plugins/environments.js) and not [lib/plugins/teams.js](https://github.com/github/safe-settings/blob/2.1.11-rc.1/lib/plugins/teams.js) were causing the problem.

My only idea is that something is failing along the way when calling ***MergeDeep*** and that somehow affects the other processes. Or poor error handling and the code is breaking somewhere and not being properly represented.

I commented and uncommented the line 4 to 5 times; the result is consistent. When that line is removed from environments.js, everything seems to be working fine. Once I add it back I see the problems again.

**Comment by gramatus** [comment]
Yeah. I made a PR with a suggested fix and some more details on what happens, #682.

**Comment by gregnrobinson** [comment]
Awesome thanks @torgst.

@decyjphr We should look to get this fix merged ASAP as anyone using `2.1.11` or above will face these issues.

**Comment by gregnrobinson** [comment]
> Yeah. I made a PR with a suggested fix and some more details on what happens, #682.

I just tested these changes in `2.1.11`...

- Adding/removing teams is working.
- Adding/removing environments is working.
- Modifying environments is working.
  - Added/removed variables
  - Added/removed reviewers
  - Enabled/Disabled prevent self-review.

It appears to resolve the issues and everything else is still operating correctly.

**Comment by gramatus** [comment]
Tested with `2.1.13`, my problem is fixed now.
